### ISSUE TITLE
[Accessibility]: Notebooks document outline 

### ISSUE THREAD

**Issue by marisak** [sourcegraph/sourcegraph#35150]
### Audit type

Mobile device navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/35142

### Problem description

There is no way to open the document outline in Notebooks on a small screen. While this behavior was intended at creation, it doesn't meet the parity criteria, and so should be changed. 

### Expected behavior

The expected behavior is a way to open the Notebooks document outline. I would be interested in validating whether or not that's needed. (Probably) 

### Additional details

![sourcegraph com_notebooks_Tm90ZWJvb2s6Mjk4](https://user-images.githubusercontent.com/16570574/167484025-4f411b59-8be0-4648-8d78-485601a72847.png)

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.

**Comment by limitedmage** [comment]
Since the sidebar is meant as extra help to use screen real estate when available, and all functionality of the sidebar is non-essential (the notebook can be scrolled instead), I think it's OK to not address this. @benvenker @almeidapaulooliveira please reopen and assign design resourcing if you disagree.
